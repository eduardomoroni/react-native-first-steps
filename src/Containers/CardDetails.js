/* @flow */

import React, { Component } from 'react'
import {
  View,
  StyleSheet,
  Image
} from 'react-native'
import {
  Metrics
} from '../Styles/Themes'

type CardDetailsProps = {
  card: any
}

export default class CardDetails extends Component {
  props: CardDetailsProps

  render () {
    const {
      multiverseid
    } = this.props.card
    const IMG_URL = `http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=${multiverseid}&type=card`

    return (
      <View style={styles.container}>
        <Image style={styles.image} source={{uri: IMG_URL}} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: Metrics.navBarHeight
  },
  image: {
    flex: 1
  }
})
