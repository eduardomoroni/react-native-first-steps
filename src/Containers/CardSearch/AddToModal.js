import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
import { cardType } from '../../Types/CardType'
import { Modal, SubmitButtonForm, DropdownInputForm } from '../Components'
import WishListService from '../../Services/WishListService'
import { reduxForm, Field, formValueSelector } from 'redux-form'
import { connect } from 'react-redux'

import { RkText } from 'react-native-ui-kitten'

export class AddToModal extends Component {
  constructor () {
    super()
    this.state = {
      isVisible: true
    }
  }

  toggleModal () {
    this.state.isVisible = !this.state.isVisible
  }

  addCardAmountToList (listName, card, amount) {
    const { wishList } = this.props
    console.log(wishList[listName], listName, amount)
    WishListService.updateCardAmount(card, 2, wishList.have)
  }

  render () {
    const {
      card,
      wishList,
      howMuchCards,
      whichTradeList
    } = this.props

    const tradeLists = ['want', 'have']
    const numberRange = ['1', '2', '3', '4']

    console.log(wishList)
    return (
      <Modal isVisible onModalHide={() => this.toggleModal()} >
        <View style={styles.modal}>
          <RkText style={styles.title}>
            {`Add to TradeList`}
          </RkText>
          <Field name='whichTradeList' component={DropdownInputForm} dropdownItems={tradeLists} selectedValue={whichTradeList} />
          <Field name='howMuchCards' component={DropdownInputForm} dropdownItems={numberRange} selectedValue={howMuchCards} />
        </View>
        <View>
          <SubmitButtonForm onPress={() => this.addCardAmountToList(whichTradeList, card, howMuchCards)} />
        </View>
      </Modal>
    )
  }
}

AddToModal.propTypes = {
  card: cardType.isRequired,
  listToAdd: PropTypes.any
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  title: {
    fontSize: 18,
    textAlign: 'center'
  },
  modal: {
    backgroundColor: 'white',
    height: 400,
    borderTopRightRadius: 4,
    borderTopLeftRadius: 4
  }
})

const mapStateToProps = (state) => {
  const selector = formValueSelector('AddToModalForm')

  return {
    whichTradeList: selector(state, 'whichTradeList'),
    howMuchCards: selector(state, 'howMuchCards')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
  }
}

const AddToModalDecorated = reduxForm({form: 'AddToModalForm'})(AddToModal)
export default connect(mapStateToProps, mapDispatchToProps)(AddToModalDecorated)
