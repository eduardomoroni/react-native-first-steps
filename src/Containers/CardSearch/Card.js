/* @flow */

import React, { PureComponent } from 'react'
import { View, Text } from 'react-native'
import styles from '../../Styles/CardStyle'
import { placeholdersToSymbols } from '../../Realm/Conversion/Placeholder'

// TODO: Make this component a dumb component
// TODO: Use prop-types
type CardProps = {
  card: any,
  showCardText: boolean
}

export const getLastPrinting = (printings: any) => {
  const keys = Object.keys(printings)
  const lastPrinting = printings[keys[keys.length - 1]]
  return lastPrinting.printing
}

export class Card extends PureComponent {
  props: CardProps

  render () {
    const { card, showCardText } = this.props
    const {
      name,
      text,
      manaCost,
      type,
      power,
      toughness,
      printings
    } = card
    return (
      <View style={styles.container}>
        {renderCardNameAndMana(name, placeholdersToSymbols(manaCost))}
        {renderCardTypeAndEdition(type, getLastPrinting(printings))}
        {renderTextAndPower(text, power, toughness, showCardText)}
      </View>
    )
  }
}

const renderCardNameAndMana = (leftText, rightText) => {
  return (
    <View style={styles.lineContainer}>
      <View style={styles.leftWord}>
        <Text key='cardName' style={styles.cardNameText}>{leftText}</Text>
      </View>
      <View style={styles.rightWord}>
        <Text key='cardManaCost' style={styles.mana}>{rightText}</Text>
      </View>
    </View>
  )
}

const renderCardTypeAndEdition = (leftText, rightText) => {
  return (
    <View style={styles.lineContainer}>
      <View style={styles.leftWord}>
        <Text key='cardType' style={styles.cardTypeText}>{leftText}</Text>
      </View>
      <View style={styles.rightWord}>
        <Text key='cardEdition'>{rightText}</Text>
      </View>
    </View>
  )
}

const renderTextAndPower = (text, power, toughness, showCardText) => {
  return (
    <View style={styles.cardTextContainer}>
      <View style={styles.cardText}>
        <Text key='cardText' style={styles.text}>{showCardText ? text : ''}</Text>
      </View>
      <View style={styles.cardPower}>
        <Text key='cardPowerAndToughness' style={styles.cardPowerToughness}>{power || toughness ? `${power}/${toughness}` : ''}</Text>
      </View>
    </View>
  )
}
