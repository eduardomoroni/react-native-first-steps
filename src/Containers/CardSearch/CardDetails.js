/* @flow */

import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'
import { Metrics } from '../../Styles/Themes'
import { CardImage } from '../Components'

type CardDetailsProps = {
  card: any
}

export default class CardDetails extends Component {
  props: CardDetailsProps

  render () {
    return (
      <View style={styles.container}>
        <CardImage card={this.props.card} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: Metrics.navBarHeight
  }
})
