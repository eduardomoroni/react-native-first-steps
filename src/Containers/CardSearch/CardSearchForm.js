/* @flow */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { View, Keyboard } from 'react-native'
import { reduxForm, Field, formValueSelector } from 'redux-form'
import { searchForCards, showFormModal } from '../../Redux/Actions'
import Styles from '../../Styles/CardSearchFormStyle'
import { valuesOf } from '../../Realm/RealmService'
import type { Dispatch } from 'redux'
import {
  DropdownInputForm,
  Modal,
  MultiSelect,
  ManaIconsBar,
  SubmitButtonForm,
  NumericInputForm,
  ModalToggle,
  TextInputForm
} from '../Components'

export class CardSearchForm extends Component {
  renderThreeFieldInRow (fieldOne, fieldTwo, fieldThree) {
    return (
      <View style={Styles.multipleFieldsPerLine}>
        <View style={Styles.leftField}>
          {fieldOne}
        </View>
        <View style={Styles.middleField}>
          {fieldTwo}
        </View>
        <View style={Styles.rightField}>
          {fieldThree}
        </View>
      </View>
    )
  }

  renderTwoFieldInRow (leftField, rightField) {
    return (
      <View style={Styles.multipleFieldsPerLine}>
        <View style={Styles.leftField}>
          {leftField}
        </View>
        <View style={Styles.rightField}>
          {rightField}
        </View>
      </View>
    )
  }

  renderModal () {
    let modalContent = {}
    let { visibleModal, cardSets, cardFormats, cardRarities } = this.props

    if (visibleModal === 'cardRarity') {
      modalContent = <Field name='cardRarity' component={MultiSelect} items={cardRarities} />
    } else if (visibleModal === 'cardSet') {
      modalContent = <Field name='cardSet' component={MultiSelect} items={cardSets} />
    } else if (visibleModal === 'cardFormat') {
      modalContent = <Field name='cardFormat' component={MultiSelect} items={cardFormats} />
    } else {
      return <View />
    }
    return (
      <Modal isVisible onModalHide={() => this.props.showModal('')} >
        {modalContent}
      </Modal>
    )
  }

  render () {
    const {
      handleSubmit,
      cardTypes,
      cardSubTypes,
      cardType,
      cardSubType,
      showModal,
      searchCards,
      cardRarity,
      cardSet,
      cardFormat
    } = this.props

    const formSubmit = (formValues) => {
      Keyboard.dismiss()
      console.log('Form sent: ', formValues)
      searchCards(formValues)
    }

    const numericOperators = ['', '<', '<=', '=', '>=', '=']

    return (
      <View style={Styles.container}>
        <View style={Styles.formContainer}>
          <Field name='cardName' component={TextInputForm} />
          <Field name='cardText' component={TextInputForm} />
          { this.renderTwoFieldInRow(
            <Field name='cardType' component={DropdownInputForm} dropdownItems={cardTypes} selectedValue={cardType} />,
            <Field name='cardSubType' component={DropdownInputForm} dropdownItems={cardSubTypes} selectedValue={cardSubType} />
        )}
          { this.renderThreeFieldInRow(
            <Field name='cardPower' component={NumericInputForm} dropdownItems={numericOperators} />,
            <Field name='cardToughness' component={NumericInputForm} dropdownItems={numericOperators} />,
            <Field name='cardCMC' component={NumericInputForm} dropdownItems={numericOperators} />
        )}
          <Field name='cardColors' component={ManaIconsBar} />
          {/* Secundary Options */}
          <Field name='cardFlavorText' component={TextInputForm} />
          { this.renderTwoFieldInRow(
            <Field name='cardArtist' component={TextInputForm} />,
            <Field name='cardCollectionNumber' component={TextInputForm} keyboardType={'numeric'} maxLength={3} />
        )}
          { this.renderThreeFieldInRow(
            <ModalToggle label='cardRarity' onPress={showModal} selected={cardRarity} />,
            <ModalToggle label='cardSet' onPress={showModal} selected={cardSet} />,
            <ModalToggle label='cardFormat' onPress={showModal} selected={cardFormat} />
        )}
          <Field name='cardColorsIdentity' component={ManaIconsBar} />
          {this.renderModal()}
        </View>
        <View style={Styles.containerFooter}>
          <SubmitButtonForm onPress={handleSubmit(formSubmit)} />
        </View>
      </View>
    )
  }
}

Field.propTypes = {
  dropdownItems: PropTypes.arrayOf(PropTypes.string),
  selectedValue: PropTypes.string
}

const mapStateToProps = (state) => {
  const selector = formValueSelector('CardSearchForm')

  const cardRarities = ['Common', 'Uncommon', 'Rare', 'Mythic Rare']
  const cardTypes = valuesOf('Type')
  const cardSubtypes = valuesOf('SubType')
  const printings = valuesOf('Printing')
  const formats = valuesOf('Legality')

  cardTypes.unshift('')
  cardSubtypes.unshift('')

  return {
    cardType: selector(state, 'cardType'),
    cardSubType: selector(state, 'cardSubType'),
    cardSubTypes: cardSubtypes,
    cardTypes: cardTypes,
    cardSets: printings,
    cardFormats: formats,
    cardRarity: selector(state, 'cardRarity'),
    cardSet: selector(state, 'cardSet'),
    cardFormat: selector(state, 'cardFormat'),
    visibleModal: state.cardSearch.visibleModal,
    cardRarities: cardRarities
  }
}

const mapDispatchToProps = (dispatch: Dispatch<*>) => {
  return {
    searchCards: (cardSearchForm) => dispatch(searchForCards(cardSearchForm)),
    showModal: (modal) => dispatch(showFormModal(modal))
  }
}

const cardSearchFormDecorated = reduxForm({form: 'CardSearchForm'})(CardSearchForm)
export default connect(mapStateToProps, mapDispatchToProps)(cardSearchFormDecorated)
