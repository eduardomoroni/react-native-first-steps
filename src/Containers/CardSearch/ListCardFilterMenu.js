// @flow

import React, { Component } from 'react'
import { View, BackAndroid, Switch, Text, Picker } from 'react-native'
import { reduxForm, Field, formValueSelector } from 'redux-form'
import { connect } from 'react-redux'
import { toggleShowCardText, sortCards as sortCardsActionCreator } from '../../Redux/Actions'
import I18n from 'react-native-i18n'

const renderSwitch = (fieldProps) => {
  const { input, isEnabled } = fieldProps
  const { onChange, name } = input

  return (
    <View style={styles.rowStyle}>
      <Text>{I18n.t(name)}</Text>
      <Switch style={styles.switchStyle} value={isEnabled} onValueChange={onChange} />
    </View>
  )
}

const renderSortInput = (fieldProps) => {
  const { onChange, name, value } = fieldProps.input

  const sortValues = ['cmc', 'multiverseid', 'name', 'number', 'rarity', 'type', 'power', 'toughness']

  const renderPickerItem = (value, key) => {
    return <Picker.Item label={value} value={value} key={key} />
  }

  return (
    <View style={styles.rowStyle}>
      <Text>{I18n.t(name)}</Text>
      <Picker
        selectedValue={value.field}
        onValueChange={onChange}>
        { sortValues.map(renderPickerItem) }
      </Picker>
    </View>
  )
}

class ListCardFilterMenu extends Component {
  componentDidMount () {
    BackAndroid.addEventListener('hardwareBackPress', () => {
      if (this.context.drawer.props.open) {
        this.context.drawer.toggle()
        return true
      }
      return false
    })
  }

  componentWillReceiveProps (nextProps) {
    const {
      showCardText,
      toggleCardText,
      sortBy,
      sortCards,
      sortReverseOrder
    } = nextProps

    if (this.props.showCardText !== showCardText) {
      toggleCardText({showCardText: showCardText})
    }
    if (this.props.sortBy !== sortBy || this.props.sortReverseOrder !== sortReverseOrder) {
      sortCards({field: sortBy, reversed: sortReverseOrder})
    }
  }

  render () {
    const { showCardText, sortReverseOrder } = this.props
    return (
      <View style={styles.container}>
        <Field name='showCardText' component={renderSwitch} isEnabled={showCardText} />
        <Field name='sortReverseOrder' component={renderSwitch} isEnabled={sortReverseOrder} />
        <Field name='sortBy' component={renderSortInput} />
      </View>
    )
  }
}

ListCardFilterMenu.contextTypes = {
  drawer: React.PropTypes.object
}

ListCardFilterMenu.propTypes = {
  showCardText: React.PropTypes.bool,
  sortBy: React.PropTypes.string,
  sortReverseOrder: React.PropTypes.bool,
  toggleCardText: React.PropTypes.func,
  sortCards: React.PropTypes.func
}

Field.propTypes = {
  isEnabled: React.PropTypes.bool
}

const mapStateToProps = (state) => {
  const selector = formValueSelector('CardSearchFilter')

  return {
    initialValues: {
      showCardText: state.cardSearch.showCardText,
      sortBy: state.cardSearch.sortBy.field,
      sortReverseOrder: state.cardSearch.sortBy.reversed
    },
    showCardText: selector(state, 'showCardText'),
    sortBy: selector(state, 'sortBy'),
    sortReverseOrder: selector(state, 'sortReverseOrder')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    toggleCardText: (filter) => dispatch(toggleShowCardText(filter)),
    sortCards: (sortParams) => dispatch(sortCardsActionCreator(sortParams))
  }
}

const styles = {
  container: {
    flex: 1,
    padding: 20
  },
  switchStyle: {
    marginBottom: 10
  },
  rowStyle: {
    height: 40
  }
}

const ListCardFilterDecorated = reduxForm({form: 'CardSearchFilter'}, mapStateToProps)(ListCardFilterMenu)
export default connect(mapStateToProps, mapDispatchToProps)(ListCardFilterDecorated)
