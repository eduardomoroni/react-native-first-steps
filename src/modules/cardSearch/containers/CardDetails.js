// @flow

import React, { Component } from 'react'
import { Text, View, ScrollView } from 'react-native'
import { styles } from '../../theme/cardSearch/CardDetailsStyle'
import { CardImage } from '../../shared/components/index'
import { cardType, rulingsType } from '../../../types/CardType'
import { rulingsTextAsArray } from '../../../services/realm/conversion/RealmRepresentation'
import I18n from 'react-native-i18n'

export default class CardDetails extends Component {
  renderText (value: string, key: number) {
    return (
      <Text key={key} >{`
        - ${value}`
      }</Text>
    )
  }

  renderRulings (rulings: rulingsType) {
    const rulesText = rulingsTextAsArray(rulings)

    return (
      <View style={styles.rowContainer}>
        <Text style={styles.rowText}>
          <Text style={styles.rowName}>{`${I18n.t('rulings')}: `}</Text>
          { rulesText.map(this.renderText) }
        </Text>
      </View>
    )
  }

  renderRow (rowName: string, rowText: string) {
    if (rowText === null) {
      return null
    }

    return (
      <View style={styles.rowContainer}>
        <Text style={styles.rowText}>
          <Text style={styles.rowName}>{`${I18n.t(rowName)}: `}</Text>
          {rowText}
        </Text>
      </View>
    )
  }

  render () {
    const { card } = this.props

    return (
      <ScrollView style={styles.container}>
        <View style={styles.card}>
          <CardImage card={card} />
        </View>
        {this.renderRow('name', card.name)}
        {this.renderRow('type', card.type)}
        {this.renderRow('P/T', `${card.power}/${card.toughness}`)}
        {this.renderRow('manaCost', `${card.manaCost} (${card.cmc})`)}
        {this.renderRow('rarity', card.rarity)}
        {this.renderRow('text', card.text)}
        {this.renderRow('flavor', card.flavor)}
        {this.renderRulings(card.rulings)}
        {this.renderRow('sets', 'Estou com pregui√ßa')}
        {this.renderRow('legality', 'Farei depois')}
        {this.renderRow('foreignNames', 'Um dia')}
        <View style={styles.card} />
      </ScrollView>
    )
  }
}

CardDetails.propTypes = {
  card: cardType
}
