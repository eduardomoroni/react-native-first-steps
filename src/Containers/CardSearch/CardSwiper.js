import React, { Component } from 'react'
import PropTypes from 'prop-types'
import CardDetails from './CardDetails'
import { Actions as NavigationActions } from 'react-native-router-flux'
import { cardType } from '../../Types/CardType'
import { getIndex } from '../../Realm/RealmService'
import Swiper from 'react-native-swiper'

class CardSwiper extends Component {
  renderCardDetails (card, key) {
    return (
      <CardDetails card={card} key={key} />
    )
  }

  updateScreenTitle (e, state) {
    const { index } = state
    const getCardName = (index) => this.children[index].props.card.name
    NavigationActions.refresh({title: `${getCardName(index)}`})
  }

  render () {
    const { cards, card } = this.props
    let { index } = this.props

    if (card !== null) {
      index = getIndex(cards, card)
    }

    return (
      <Swiper
        ref={(component) => { this.swiper = component }}
        onMomentumScrollEnd={this.updateScreenTitle}
        style={{ flex: 1 }}
        loadMinimal
        loadMinimalSize={2}
        loop={false}
        showsPagination={false}
        index={index}
        >
        {cards.map(this.renderCardDetails)}
      </Swiper>
    )
  }
}

CardSwiper.propTypes = {
  cards: PropTypes.objectOf(cardType),
  card: cardType,
  index: PropTypes.number
}

CardSwiper.defaultProps = {
  index: 0,
  card: null
}

export default CardSwiper
