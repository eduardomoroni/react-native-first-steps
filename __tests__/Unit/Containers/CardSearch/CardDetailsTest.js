import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { CardDetails } from '../../../../src/Containers/CardSearch/CardDetails'

const props = {
  card: {
    TDB: 'To de defined'
  }
}

describe('<CardDetails />', () => {
  it('Should Render CardDetails', () => {
    const wrapper = shallow(<CardDetails {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
