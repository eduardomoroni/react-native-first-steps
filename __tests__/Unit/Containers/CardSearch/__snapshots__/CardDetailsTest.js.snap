// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardDetails /> Should Render CardDetails 1`] = `
<ScrollViewMock
  style={
    Object {
      "flex": 1,
      "paddingTop": 64,
    }
  }
>
  <View
    style={
      Object {
        "height": 1036,
        "marginHorizontal": 5,
        "width": 740,
      }
    }
  >
    <CardImage
      card={
        Object {
          "artist": "Zack Stella",
          "cmc": 2,
          "colorIdentity": Object {
            "0": Object {
              "colorIdentity": "U",
            },
          },
          "colors": Object {
            "0": Object {
              "color": "Blue",
            },
          },
          "flavor": "It is fascinating to see how the people of this plane replicate magical techniques without a natural gift.",
          "foreignNames": Object {
            "0": Object {
              "language": "Chinese Simplified",
              "multiverseid": 423891,
              "name": "失效",
            },
            "1": Object {
              "language": "Chinese Traditional",
              "multiverseid": 424075,
              "name": "失效",
            },
            "2": Object {
              "language": "French",
              "multiverseid": 424443,
              "name": "Négation",
            },
            "3": Object {
              "language": "German",
              "multiverseid": 424259,
              "name": "Negieren",
            },
            "4": Object {
              "language": "Italian",
              "multiverseid": 424627,
              "name": "Negare",
            },
            "5": Object {
              "language": "Japanese",
              "multiverseid": 424811,
              "name": "否認",
            },
            "6": Object {
              "language": "Korean",
              "multiverseid": 424995,
              "name": "부인",
            },
            "7": Object {
              "language": "Portuguese (Brazil)",
              "multiverseid": 425179,
              "name": "Negar",
            },
            "8": Object {
              "language": "Russian",
              "multiverseid": 425363,
              "name": "Отвергнуть",
            },
            "9": Object {
              "language": "Spanish",
              "multiverseid": 425547,
              "name": "Negar",
            },
          },
          "id": "66106dba089787a1d0d5fe1b80091e7eebe29e55",
          "imagename": "negate",
          "layout": "normal",
          "legalities": Object {
            "0": Object {
              "format": "Battle for Zendikar Block",
              "legality": "Legal",
            },
            "1": Object {
              "format": "Commander",
              "legality": "Legal",
            },
            "2": Object {
              "format": "Khans of Tarkir Block",
              "legality": "Legal",
            },
            "3": Object {
              "format": "Legacy",
              "legality": "Legal",
            },
            "4": Object {
              "format": "Lorwyn-Shadowmoor Block",
              "legality": "Legal",
            },
            "5": Object {
              "format": "Modern",
              "legality": "Legal",
            },
            "6": Object {
              "format": "Standard",
              "legality": "Legal",
            },
            "7": Object {
              "format": "Vintage",
              "legality": "Legal",
            },
          },
          "manaCost": "{1}{U}",
          "multiverseid": 423707,
          "name": "Negate",
          "number": "40",
          "originalText": "Counter target noncreature spell.",
          "originalType": "Instant",
          "power": null,
          "printings": Object {
            "0": Object {
              "printing": "pMPR",
            },
            "1": Object {
              "printing": "MOR",
            },
            "10": Object {
              "printing": "ORI",
            },
            "11": Object {
              "printing": "OGW",
            },
            "12": Object {
              "printing": "CN2",
            },
            "13": Object {
              "printing": "AER",
            },
            "2": Object {
              "printing": "M10",
            },
            "3": Object {
              "printing": "DPA",
            },
            "4": Object {
              "printing": "M11",
            },
            "5": Object {
              "printing": "M12",
            },
            "6": Object {
              "printing": "M13",
            },
            "7": Object {
              "printing": "M14",
            },
            "8": Object {
              "printing": "M15",
            },
            "9": Object {
              "printing": "DTK",
            },
          },
          "rarity": "Common",
          "rulings": Object {
            "0": Object {
              "date": "2009-10-01",
              "text": "A creature spell is any spell with the type creature, even if it has other types such as artifact or enchantment. Older cards of type summon are also creature spells.",
            },
            "1": Object {
              "date": "2009-10-01",
              "text": "Testing.",
            },
          },
          "subtypes": Object {},
          "superTypes": Object {},
          "text": "Counter target noncreature spell.",
          "toughness": null,
          "type": "Instant",
          "types": Object {
            "0": Object {
              "type": "Instant",
            },
          },
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginHorizontal": 10,
        "marginVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "MTGX",
          "fontSize": 17,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I18n.t(name): 
      </Text>
      Negate
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginHorizontal": 10,
        "marginVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "MTGX",
          "fontSize": 17,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I18n.t(type): 
      </Text>
      Instant
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginHorizontal": 10,
        "marginVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "MTGX",
          "fontSize": 17,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I18n.t(P/T): 
      </Text>
      null/null
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginHorizontal": 10,
        "marginVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "MTGX",
          "fontSize": 17,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I18n.t(manaCost): 
      </Text>
      {1}{U} (2)
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginHorizontal": 10,
        "marginVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "MTGX",
          "fontSize": 17,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I18n.t(rarity): 
      </Text>
      Common
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginHorizontal": 10,
        "marginVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "MTGX",
          "fontSize": 17,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I18n.t(text): 
      </Text>
      Counter target noncreature spell.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginHorizontal": 10,
        "marginVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "MTGX",
          "fontSize": 17,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I18n.t(flavor): 
      </Text>
      It is fascinating to see how the people of this plane replicate magical techniques without a natural gift.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginHorizontal": 10,
        "marginVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "MTGX",
          "fontSize": 17,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I18n.t(rulings): 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        
                - A creature spell is any spell with the type creature, even if it has other types such as artifact or enchantment. Older cards of type summon are also creature spells.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        
                - Testing.
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginHorizontal": 10,
        "marginVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "MTGX",
          "fontSize": 17,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I18n.t(sets): 
      </Text>
      Estou com preguiça
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginHorizontal": 10,
        "marginVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "MTGX",
          "fontSize": 17,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I18n.t(legality): 
      </Text>
      Farei depois
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginHorizontal": 10,
        "marginVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "MTGX",
          "fontSize": 17,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I18n.t(foreignNames): 
      </Text>
      Um dia
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 1036,
        "marginHorizontal": 5,
        "width": 740,
      }
    }
  />
</ScrollViewMock>
`;
